lua/rocks-config/init.lua

```diff
@@ -205,7 +205,8 @@ function rocks_config.setup(user_configuration)
     end
 
     for name, data in pairs(user_configuration.plugins or {}) do
-        if exclude[name] then
+        -- If plugin is in a bundle with a config or has no_setup = true do not attempt to autoinvoke the setup() function
+        if exclude[name] or data.no_setup then
```

Comment:
note: I know I initially suggested adding a `nosetup` field in https://github.com/nvim-neorocks/rocks-config.nvim/issues/9#issue-2058988695, but:

Come to think of it, I believe a individual `auto_setup` field would be better. That way, you could enable `auto_setup` globally and override it by setting it to `false` for individual plugins. Or, you could disable it globally and enable it for individual plugins.

For another time: Apart from being a boolean, it could also be a string, which specifies the module containing the `setup` function (e.g. `"nvim-treesitter.configs"`).

@vhyrro what do you think?

Reply:
On the other hand, given https://github.com/nvim-neorocks/rocks-config.nvim/issues/9#issuecomment-2156496171, what if a plugin has a `config` but `auto_setup` is disabled for that plugin?
That's technically an invalid config.

Reply:
I was thinking about this too, I agree that `no_setup` isn't the best name here. My thought was it would be nice to name it just `setup`. I like this because it could be used alongside `auto_setup` by setting a plugin's field `setup = false` or it could be used without `auto_setup` with `setup = true` calling setup.

The problem I see is this kind of clashes with what the `config` field does now. This could be resolved by having `config` be for config objects only, with an inferred `setup = true` when it is provided. Then move the functionality of `config = 'path-to-config'` to be under a string/boolean `setup` field.

Reply:
We shouldn't rename `config` to `setup`.

- That would be a breaking change.
-  `config` conveys more clearly that it is primarily meant for configuration. Automatically calling a `setup` function is just a fallback if `auto_setup` is enabled.
- The `setup` function [is an antipattern](https://mrcjkb.dev/posts/2023-08-22-setup.html). I fear renaming `config` to `setup` could further encourage its use.

Reply:
My intent is not to rename `config`. Here I have added `setup` alongside `config` and rescoped `config` to be for table only. I understand this would be breaking as I put in my other comment, so I am fine to rework this and support a string value for both fields if needed.

Alternatively we could add option to provide a boolean value for `config`.

This would work well with how `rocks-config.nvim` is currently set up, but I do think it packs a lot into a single field. A `config` could then be a string (module path), table (options to provide to `setup()`, or a boolean (enable/disable calling `setup`).

I know you previously recommended naming the field `auto_setup`. I personally am not a big fan of this. For user understanding I think it is best if `auto_setup` refers to the global setting only.

Reply:
> Alternatively we could add option to provide a boolean value for config instead of creating an additional field for setup or auto_setup.
>
> This would work well with how rocks-config.nvim is currently set up, but I do think it packs a lot into a single field. A config could then be a string (module path), table (options to provide to setup(), or a boolean (enable/disable calling setup).

I think this would be the best option, as it avoids the possibility of an invalid state entirely. 
I don't mind it "packing a lot into a single field" as long as it's clearly documented (it's just a union type, which is fine).

But I would want to wait and see what @vhyrro has to say.